### ISSUE TITLE
Replication inconsistent if both replicate-to and replicate-to-matching are used

### ISSUE THREAD

**Issue by steinri** [mittwald/kubernetes-replicator#91]
**Describe the bug**
If I use both annotations `replicator.v1.mittwald.de/replicate-to` and `replicator.v1.mittwald.de/replicate-to-matching` it sometimes replicates and sometimes does not replicate.

The behavior is that it gets replicated when a namespaces is created with the label, but updates are not propagated (neither if I update the namespace nor if I update the secret).

**To Reproduce**

First I create a new name secret, which should be replicated into the namespace `foobar` and also into all namespaces with the `foo: bar` label.

```yaml
---
apiVersion: v1
kind: Secret
metadata:
  name: my-test-secret
  annotations:
    replicator.v1.mittwald.de/replicate-to: "^foobar$"
    replicator.v1.mittwald.de/replicate-to-matching: foo=bar
data:
  key1: dmFsdWUx
```

Then I create a namespace with the `foo: bar` label:

```yaml
---
apiVersion: v1
kind: Namespace
metadata:
  labels:
    foo: bar
  name: test-namespace-1
```

As expected the secret appears:

```bash
$ kubectl get -n test-namespace-1 secrets my-test-secret -oyaml
apiVersion: v1
data:
  key1: dmFsdWUx
kind: Secret
# ...
```

**Bug 1**
Now I want to change value in the secret

```yaml
---
apiVersion: v1
kind: Secret
metadata:
  name: my-test-secret
  annotations:
    replicator.v1.mittwald.de/replicate-to: "^foobar$"
    replicator.v1.mittwald.de/replicate-to-matching: foo=bar
data:
  key1: dmFsdWUy
```

However the *replicated secret does not change*:

```bash
$ kubectl get -n test-namespace-1 secrets my-test-secret -oyaml
apiVersion: v1
data:
  key1: dmFsdWUx
kind: Secret
# ...
```

**Bug 2**

Now I first create the namespace

```yaml
---
apiVersion: v1
kind: Namespace
metadata:
  name: test-namespace-2
```

and then add the label

```yaml
---
apiVersion: v1
kind: Namespace
metadata:
  labels:
    foo: bar
  name: test-namespace-2
```

The *secret does not appear in that namespace*:

```bash
$ kubectl get -n test-namespace-2 secrets my-test-secret -oyaml
Error from server (NotFound): secrets "my-test-secret" not found
```

**Expected behavior**

A. I would expect the secret to be replicated into the matching namespaces as if the `replicator.v1.mittwald.de/replicate-to` was not set in addition.
B. If they are mutually exclusive it should be clear in the documentation and the replication should not work in the create namespace case.

**Environment:**
 - Kubernetes version: 1.19
 - kubernetes-replicator version: v2.5.1

**Additional context**

**Comment by YannikBramkamp** [comment]
Thank you for the report! I tried to reproduce this issue with the provided instructions, but unfortunately I was unable to observe the behavior you described. In my test-cluster the secret wouldn't get replicated at all with both annotations set upon creation of a new labelled namespace. 

However, after internal discussion we concluded that either way the behavior isn't intuitive. I will therefore look into this further when I have the time and explore the option to possibly enable both annotations to be set in the same resource.
